ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/validators.tsx

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+const usng = require('usng.js')
+const converter = new usng.Converter()
+const errorMessages = require('../../component/location-new/utils/errors')
+import { validateGeo } from '../utils/validation'
+
+const dmsRegex = new RegExp('^([0-9_]*)°([0-9_]*)\'([0-9_]*\\.?[0-9_]*)"$')
+
+type Direction = 'N' | 'S' | 'E' | 'W'
+
+type Point = {
+  latDirection: Direction
+  lonDirection: Direction
+  lat: string
+  lon: string
+}
+
+type DmsCoordinate = {
+  degrees: string
+  minutes: string
+  seconds: string
+  direction: Direction
+}
+
+type UtmUpsPoint = {
+  easting: number
+  northing: number
+  zoneNumber: number
+  hemisphere: 'NORTHERN' | 'SOUTHERN'
+}
+
+function isUPS(input: string) {
```

Comment:
📝  This is a corner case validation for usng around the poles, when we get to the poles, we fall back to UPS so the textfields for usng/mgrs _may_ have UPS values. This tries to validate them by seeing if the value is deserializable as UPS which goes ahead and checks that it matches valid UPS regex.